packages/url_launcher/url_launcher/android/src/main/AndroidManifest.xml

```diff
@@ -5,4 +5,10 @@
       android:theme="@android:style/Theme.NoTitleBar.Fullscreen"
       android:exported="false"/>
   </application>
+  <queries>
+    <intent>
+      <action android:name="android.intent.action.SENDTO" />
```

Comment:
This was causing an issue in Android 11

Reply:
For which scheme? It sounds like we may need to update [the README](https://github.com/flutter/plugins/blob/master/packages/url_launcher/url_launcher/README.md#android).

Reply:
Android manifests are aggregated. There's no need for manual instructions.

Reply:
SENDTO works for SMS and emails

Reply:
> Android manifests are aggregated. There's no need for manual instructions.

Oh, I thought this was in the example app. I don't think we should add this here, because we don't know what schemes someone is going to use. We should instead document this query in the README for people who need it, as we are currently doing (but without this scheme). And then add this to the example instead, so that the test passes.

Reply:
Done. I don't know why not adding them to the plugin AndroidManifest.xml automatically. I think the idea is to have 3p dependencies auto-configuring themselves when possible unless there's some major concern that prevents these from being the default.

I'm not sure if these concerns exist, or if we didn't know that the Android Gradle plugin merges manifests automatically.

Reply:
> I don't know why not adding them to the plugin AndroidManifest.xml automatically.

It is entirely possible to use `url_launcher` without any permissions (just use `launch` without `canLaunch`), and almost nobody would use the entire set of possible queries that `canLaunch` could require. We shouldn't force people using a plugin to have their app require permissions they aren't using. Especially since the guidance from the stores is always to use the minimal permission set.

> if we didn't know that the Android Gradle plugin merges manifests automatically

I didn't know that certainly, and it's good to know 🙂 . We should audit our READMEs for cases where the advice is "you must add X if you use this plugin" rather than "you might need X if you are using this plugin to do Y", and for the former, put it in the plugin manifest.

Reply:
I haven’t used this plugin, but if canLaunch isn’t required, then the API seems very confusing. I assumed that canLaunch means that if it returns false, then never launch the intent, and handle it differently.

Reply:
Updating the docs is on my TODO list, because the example shown in the README (`await canLaunch(_url) ? await launch(_url) : throw 'Could not launch $_url';`) is pointless; it's just a slower and more cumbersome way of writing
```
if (!(await launch(_url))) throw 'Could not launch $_url';
```

> and handle it differently

This is where `canLaunch` is actually useful, yes. But with `https` URLs, for instance, I would be willing to bet that nobody has fallback code in their application that handles the case where someone's device *does not have a browser*, so calling `canLaunch` in that cirumstance is pointless.